### PR TITLE
API doc improvements to user code in steps/*.py

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

API doc improvements to user code in steps/*.py

## How is this patch tested?

Manual inspection of docs

## Release Notes

API doc improvements to user code in steps/*.py

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [X] `area/regression`: Sklearn regression pipeline example
- [ ] `area/build`: Build and test infrastructure for MLflow pipeline example

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by jinzhang21 — APPROVED** [review]

File: steps/custom_metrics.py [hunk 1]
```diff
@@ -1,9 +1,45 @@
+"""
+This module defines custom metric functions that are invoked during the 'train' and 'evaluate'
+steps to provide model performance insights. Custom metric functions defined in this module are
+referenced by the ``function`` attributes of entries in the ``custom`` subsection of the ``metrics``
+section in ``pipeline.yaml``, for example:
```
Comment (apurva-koti) [inline steps/custom_metrics.py]: This is a bit hard to follow. Is this technically correct?

```suggestion
referenced by the ``function`` attributes of entries in the ``metrics.custom``
section in ``pipeline.yaml``, for example:
```

File: steps/train.py [hunk 2]
```diff
@@ -1,6 +1,15 @@
+"""
+This module defines the following routines used by the 'train' step of the regression pipeline:
+
+- ``estimator_fn``: Defines the customizable estimator architecture and parameters that are used
```
Comment (apurva-koti) [inline steps/train.py]: IMO this is a bit vague if I wasn't fully in context. The estimator_fn is meant to just return an estimator that implements a certain spec right? What does "architecture" refer to here?

**Review by apurva-koti — COMMENTED** [review]
Two small comments but looks good!

File: steps/train.py (hunk 2 above)
Comment (dbczumar) [reply steps/train.py]: architecture = model structure. Thanks for pointing this out - turns out the term is mostly used in deep learning & not so much in classical models. I've changed this to "estimator type and parameters." Let me know if you like that more :)

File: steps/custom_metrics.py (hunk 1 above)
Comment (dbczumar) [reply steps/custom_metrics.py]: I think the `.` notation might be confusing, so I just changed this to

```
referenced in the ``metrics`` section of ``pipeline.yaml``, for example:
```

The example should make pretty clear what we're talking about it

Comment (apurva-koti) [reply steps/custom_metrics.py]: Yep I thought that as well. Thanks!

File: steps/train.py (hunk 2 above)
Comment (apurva-koti) [reply steps/train.py]: That's great, thanks for the change :)

**Review by apurva-koti — APPROVED** [review]
LGTM!
